content/docs/1.6.0/advanced-resources/snapshot-space-management.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Snapshot Space Management
+weight: 4
+---
+
+Longhorn supports setting limitation of maximum snapshout count and total size for each volume.
+
+## Introduction
+
+Before Longhorn v1.6.0, the maximum snapshot count is hard coded to 250 in the system. It is not configurable. Users cannot limit snapshot space usage. After Longhorn v1.6.0, users can set the maximum snapshot count and total size for each volume. When the number of snapshots or the total size of snapshots exceeds the limitation, users need to delete some snapshots to make room for new snapshots.
+
+In longhorn-engine, there are different kind of usage. The system will ignore removed snapshots, backing image and volume head usage.
+
+## Settings
+
+### Global Settings
+
+- **snapshot-max-count** <br>
+
+    This setting allows users to set default maximum snapshot count. If users create a volume without `.Spec.SnapshotMaxCount`, the system will the value from the `snapshot-max-count` setting. Changing the value of `snapshot-max-count` setting will not affect the existing volumes.
+
+    The valid value should be between `2` to `250`. The default value is `250`. The system needs at least `2` snapshots to work properly. If a volume needs rebuilding, the system needs one snapshot space to merge current snapshots as one snapshot. Rebuilding also creates another snapshot, so we need at least `2` snapshot space.
+
+### Per-Volume Settings
+
+- **SnapshotMaxCount** <br>
+
+    Longhorn also supports the per-volume setting by configuring `Volume.Spec.SnapshotMaxCount`. The value is `0` by default, so snapshot count limitation is determined by the global setting `snapshot-max-count`. `Volume.Spec.SnapshotMaxCount` supports `0` and any positive integer between `2` to `250`. Each volume can have its snapshot count limitation customized.
+
+- **SnapshotMaxSize** <br>
+
+    This setting allows users to set the maximum total size of snapshots for each volume. The value is `0` by default, so snapshot size limitation is disabled. `Volume.Spec.SnapshotMaxSize` supports `0` and positive integer which is larger than `Volume.Spec.Size * 2`. The reason is that the system needs at least `2` snapshots to work properly, so we don't want `Volume.Spec.SnapshotMaxSize` blocks it. Each volume can have its snapshot size limitation customized.
+
+    If users expand volume size, the system will automatically increase the snapshot size limitation to `Volume.Spec.Size * 2` if the current snapshot size limitation is smaller than `Volume.Spec.Size * 2`.
```

Comment:
```suggestion
**SnapshotMaxSize**: Maximum aggregate size of snapshots for a specific volume. 

You can specify "0" or any value larger than `Volume.Spec.Size` multiplied by 2. You must double the value of `Volume.Spec.Size` because Longhorn requires at least two snapshots to function properly.

The default value is "0", which effectively disables the setting.

When you expand the volume size, Longhorn automatically increases the value of this setting to `Volume.Spec.Size` multiplied by 2 (if the current value is smaller).
```